user_tools/src/spark_rapids_tools/configuration/distributed_tools_config.py

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" Configuration file for distributed tools """
+from typing import List
+
+from pydantic import BaseModel, Field
+
+
+class SparkProperty(BaseModel):
+    """Represents a single Spark property with a name and value."""
+    name: str = Field(
+        description='Name of the Spark property, e.g., "spark.executor.memory".')
+    value: str = Field(
+        description='Value of the Spark property, e.g., "4g".')
+
+
+class DistributedToolsConfig(BaseModel):
+    """Configuration class for distributed tools"""
+    hdfs_output_dir: str = Field(
+        description='HDFS output directory where the output data from the distributed '
+                    'tools will be stored.',
+        examples=['hdfs:///path/to/output/dir']
+    )
+
+    spark_properties: List[SparkProperty] = Field(
```

Comment:
Technically, `spark_property` configuration is not limited to the distributed mode.
For the localMode, there is potential use to have a customer set spark_property that can be loaded by the tools. For example, `FileSystem` related arguments to access the eventlogs.